src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
 }
 
 export class Triangle {
+  public shape: Shape = 'triangle';
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - this.a)
+     * (p - this.b) * (p - this.c)) * 100) / 100;
+  }
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const perimetr = this.a + this.b + this.c;
```

Comment:
Inside constructor you can use it without  `this`
```suggestion
    const perimetr = a + b + c;
```